Bottom: 55d747949e52937b82271d649e11920b4b534b80
Top:    8c81211cf6460c594428b52aa1831c82fb9ccf89
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-01-20 13:45:15 +0800

vhost: let vhost_signal() returns whether or not

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index e75c2e2..4d6ac1a 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1532,9 +1532,6 @@ void vhost_signal(struct vhost_dev *dev, struct vhost_virtqueue *vq)
 	int left;
 	ktime_t now;
 
-	if (!vq->call_ctx)
-		return;
-
 	if (can_coalesce) {
 		now = ktime_get();
 		left = vq->coalesce_usecs -
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index faa0dd7..de8b2b5 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -165,7 +165,7 @@ void vhost_add_used_and_signal(struct vhost_dev *, struct vhost_virtqueue *,
 			       unsigned int id, int len);
 void vhost_add_used_and_signal_n(struct vhost_dev *, struct vhost_virtqueue *,
 			       struct vring_used_elem *heads, unsigned count);
-void vhost_signal(struct vhost_dev *, struct vhost_virtqueue *);
+bool vhost_signal(struct vhost_dev *, struct vhost_virtqueue *);
 void vhost_disable_notify(struct vhost_dev *, struct vhost_virtqueue *);
 bool vhost_enable_notify(struct vhost_dev *, struct vhost_virtqueue *);
